### PR TITLE
prosemirror-model@1.16.0: add whitespace option to NodeSpec

### PR DESCRIPTION
**Pull request by philippkuehn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/ProseMirror/prosemirror-model/commit/b9085fbf195e07f336c48647a4b318c8d6133054
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/prosemirror-model/index.d.ts [hunk 1]
```diff
@@ -671,6 +671,10 @@ declare class ProsemirrorNode<S extends Schema = any> {
      * uneditable [node view](#view.NodeView)).
      */
     isAtom: boolean;
+    /**
+     * The node type's [whitespace](#view.NodeSpec.whitespace) option.
+     */
+    whitespace: 'pre' | 'normal';
```
Comment (AnthonyWeston) [inline types/prosemirror-model/index.d.ts:677]: This is a [getter](https://github.com/ProseMirror/prosemirror-model/commit/b9085fbf195e07f336c48647a4b318c8d6133054#diff-76078cd7448ebeb6bb584d32f1ca27b7d2001d779ab0b68f956fbcb8791a473c), so I'd make this property `readonly`. Other than that it looks good to me.

**Review by AnthonyWeston — CHANGES_REQUESTED** [review]

File: types/prosemirror-model/index.d.ts (hunk 1 above)
Comment (philippkuehn) [reply types/prosemirror-model/index.d.ts:677]: None of the other existing properties like `isInline`, `isTextblock`, `isLeaf`, `isAtom` is marked as `readonly`. That's why I didn't do that either. That should probably be done in a separate PR for all properties then.

**Review by Narretz — CHANGES_REQUESTED** [review]
The comment at the top is quite outdated. Would be a good time to update it (as the whitespace option was intro'd in 1.16).

Should be "Type definitions for prosemirror-model 1.16"

**Comment by philippkuehn** [comment]
@Narretz What do you mean with `comment at the top`? The title of the PR?

**Comment by Narretz** [comment]
> @Narretz What do you mean with `comment at the top`? The title of the PR?

I mean the comment at the top of the file which indicates for which prosemirror-model version the types are

**Comment by philippkuehn** [comment]
@Narretz Ah, didn’t see that. Thanks and fixed!

**Review by Narretz — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]

**Comment by philippkuehn** [comment]
Ready to merge

**Comment by raulpopadineti** [comment]
@philippkuehn seems like this breaks the compatibility with `prosemirror-state`s `NodeSpec` because it's missing the new `whitespace` option when trying to create a fragment from the editor's schema inside the `onUpdate` callback from the `useEditor`.

```TS
const editor = useEditor({
  ...
  onUpdate: ({ editor }) => onUpdate(editor),
```

```TS
import { Editor } from "@tiptap/core";
import { DOMSerializer } from "prosemirror-model";

const onUpdate = (editor: Editor) => {
  const fragment = DOMSerializer.fromSchema(editor.schema).serializeFragment(editor.state.doc.content);
  ...
```

```TS
 Argument of type 'import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").Schema<any, any>' is not assignable to parameter of type 'import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").Schema<any, any>'.
  The types of 'spec.nodes' are incompatible between these types.
    Type 'import("./node_modules/@types/orderedmap/index")<import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").NodeSpec> | { ...; }' is not assignable to type 'import("./node_modules/@types/orderedmap/index")<import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").NodeSpec> | { ...; }'.
      Type 'OrderedMap<NodeSpec>' is not assignable to type 'OrderedMap<NodeSpec> | { [x: string]: NodeSpec; }'.
        Type 'import("./node_modules/@types/orderedmap/index")<import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").NodeSpec>' is not assignable to type 'import("./node_modules/@types/orderedmap/index")<import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").NodeSpec>'.
          Type 'import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").NodeSpec' is not assignable to type 'import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").NodeSpec'.
            Types of property 'toDOM' are incompatible.
              Type '((node: import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").Node<any>) => import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").DOMOutputSpec) | null | undefined' is not assignable to type '((node: import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").Node<any>) => import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").D...'.
                Type '(node: import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").Node<any>) => import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").DOMOutputSpec' is not assignable to type '(node: import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").Node<any>) => import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").DO...'.
                  Types of parameters 'node' and 'node' are incompatible.
                    Property 'whitespace' is missing in type 'import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model/index").Node<any>' but required in type 'import("./node_modules/@tiptap/core/node_modules/@types/prosemirror-model/index").Node<any>'.
    428 | }: Props) => {
    429 |   const onUpdate = (editor: Editor) => {
  > 430 |     const fragment = DOMSerializer.fromSchema(editor.schema).serializeFragment(editor.state.doc.content);
```

**Comment by Narretz** [comment]
@raulpopadineti if you look at the imports, types/prosemirror-state is pulling in its own version of types/prosemirror-model, but it should use the types/prosemirror-model from tiptap. This happens sometimes when you update prosemirror packages, I think because the dependency restrictions are too loose.

What happens if you delete `node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model`? Does the error go away?

**Comment by philippkuehn** [comment]
I also think a fresh install of your dependencies should help.

**Comment by raulpopadineti** [comment]
> @raulpopadineti if you look at the imports, types/prosemirror-state is pulling in its own version of types/prosemirror-model, but it should use the types/prosemirror-model from tiptap. This happens sometimes when you update prosemirror packages, I think because the dependency restrictions are too loose.
> 
> What happens if you delete `node_modules/@tiptap/core/node_modules/@types/prosemirror-state/node_modules/@types/prosemirror-model`? Does the error go away?

@Narretz ah, you're right. But, this might fix the issue only for me because the `node_modules` dir is automatically generated by the `yarn install` command, right? (We have it in `.gitignore`). Everyone else working on the project will still face the issue. 

For now, we've decided to stay on the `@tiptap/starter-kit@2.0.0-beta.164`, which doesn't use this latest `@types/prosemirror-model`.

**Pull request merged**